web/__mocks__/cart-products.mock.ts

```diff
@@ -1,18 +1,49 @@
+import { SanityImageSource } from '@sanity/image-url/lib/types/types';
+import { SanityImageSource as SanityImageSourceAsset } from '@sanity/asset-utils';
+
 export default [
   {
+    _id: '1',
     name: 'test 1',
-    price: 1,
-    stock: 1,
-    description: 'test 1 desc',
-    tag: 't',
+    price: -1,
+    discountPrice: -2,
+    stock: 0,
+    sizeChart: [
+      {
+        size: '1',
+        stock: 1,
+      },
+    ],
+    description: 'string',
+    images: [
+      {
+        image: null as SanityImageSource,
+        asset: null as SanityImageSourceAsset,
+      },
+    ],
+    tag: 'string',
     quantity: 1,
   },
   {
+    _id: '2',
     name: 'test 2',
-    price: 1,
-    stock: 1,
-    description: 'test 2 desc',
-    tag: 't',
+    price: -1,
+    discountPrice: -2,
```

Comment:
I am not convinced that we should save everything. I mean most of the properties are not being used. I think we should map the object and only save what we need, something like { id, name, image + asset, quantity, price, size }. It is unnecessary to save all images, separate discount price/price, description, size chart, stock, tag.

Reply:
Also if we do it like that you will have to put extra logic to consider the discount price instead of the regular price. if we map an object that logic will be done in the product detail when adding an item to the cart, the cart does not have to worry about anything because all logic about prices and the stock has already been verified.

Reply:
since adding things to the cart is not in the scope of your task just define a model that works for you and pretend that we are already adding them to the cart in that way. We will take care of the mapping and saving in the storage when we add the integration between the product detail and the cart (adding a product to the cart)